scripts/update-component-sha.sh

```diff
@@ -1,10 +1,66 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# can be run in 2 modes:
+# 1- $0 --hash $oldhash $newhash              - replace all instances of $oldhash with $newhash
+# 2- $0 --image $image $newhash        - replace all hashes for $image with $newhash
+#
+# for backwards compatibility if no argtype is given, the first method is assumed. Thus the following are equivalent:
```

Comment:
I think "By default the first method is assumed" is a correct description of the behaviour. If you were to remove it like Rolf suggested I think you would also want to remove the code which implements the default (NB: I don't think you should do that).